raw_templates.go

```diff
@@ -1,3 +1,3 @@
e( 'DOMAIN_CURRENT_SITE', 'localhost' );\n}\nif(!defined('PATH_CURRENT_SITE')) {\n    define( 'PATH_CURRENT_SITE', '/' );\n}\nif(!defined('SITE_ID_CURRENT_SITE')) {\n    define( 'SITE_ID_CURRENT_SITE', 1 );\n}\nif(!defined('BLOG_ID_CURRENT_SITE')) {\n    define( 'BLOG_ID_CURRENT_SITE', 1 );\n}\n", "templates/setup/external.sh.tmpl": "#!/bin/sh\nset -e\n\n##\n## This code will be run during setup, OUTSIDE the container.\n##\n## Because `whippet` running inside the container wouldn't be able to connect\n## to private repositories.\n##\n\nif test -f whippet.json; then\n  whippet deps install\nfi\n", "templates/docker-compose.yml.tmpl": "version: \"3\"\n\nvolumes:\n  mysql_data_{{.Name}}:\n\nservices:\n  mailcatcher:\n    image: schickling/mailcatcher\n    ports:\n      - \"1080:1080\"\n\n  beanstalk:\n    image: schickling/beanstalkd\n    ports:\n      - \"11300:11300\"\n\n  beanstalkd_console:\n    image: agaveapi/beanstalkd-console\n    ports:\n      - \"2080:80\"\n    environment:\n      BEANSTALKD_HOST: beanstalk\n      BEANSTALKD_PORT: 11300\n\n  mysql:\n    image: mariadb:10\n    ports:\n      - \"3306:3306\"\n    volumes:\n      - mysql_data_{{.Name}}:/var/lib/mysql\n    environment:\n      MYSQL_DATABASE: wordpress\n      MYSQL_ROOT_PASSWORD: foobar\n\n  wordpress:\n    image: thedxw/wpc-wordpress\n    ports:\n      - \"80:80\"\n    links:\n      - mysql\n      - mailcatcher\n      - beanstalk\n    volumes:\n      - .:/usr/src/app\n      - ./wp-content:/var/www/html/wp-content\n", "templates/script/setup.tmpl": "#!/bin/sh\nset -e\n\nscript/bootstrap\ndocker-compose exec wordpress /usr/src/app/setup/internal.sh\n", "templates/script/update.tmpl": "#!/bin/sh\nset -e\n\nscript/bootstrap\n", "templates/bin/console.tmpl": "#!/bin/sh\nset -e\n\necho '------------------------------------------------------------------------------'\necho 'DEPRECATED'\necho 'This command is deprecated and will be removed in a future version. Please use'\necho 'script/console instead.'\necho '------------------------------------------------------------------------------'\n\nexec script/console\n", "templates/bin/setup.tmpl": "#!/bin/sh\nset -e\n\necho '------------------------------------------------------------------------------'\necho 'DEPRECATED'\necho 'This command is deprecated and will be removed in a future version. Please use'\necho 'script/setup instead.'\necho '------------------------------------------------------------------------------'\n\nexec script/setup\n", "templates/bin/wp.tmpl": "#!/bin/sh\nset -e\n\nFLAGS=\n\n# Add -t flag iff STDIN is a TTY\nif [ -t 0 ]; then\n  FLAGS=-t\nfi\n\nCONTAINER=`docker-compose ps -q wordpress`\n\n# We can't use docker-compose here because docker-compose exec is equivalent\n# to docker exec -ti and docker-compose exec -T is equivalent to\n# docker exec. There is no docker-compose equivalent to docker exec -i.\n#\n# Issue: https://github.com/docker/compose/issues/3352\n\ndocker exec -i ${FLAGS} ${CONTAINER} wp \"${@}\"\n"}
```

Comment:
What's the change here? The very long line is very hard to parse.

Reply:
This file is compiled from the files in `templates/` (so the change here is essentially a compiled version of the change in `templates/script/server.tmpl`). We should probably mark it as a binary in `.gitattributes` to make that clear.

Reply:
I'm not a fan of using `.gitattributes` to hide changes. Making the generator add something to this file that says that it's generated would help, though.

Reply:
This is a compiled file. Its contents are generated from the contents of `templates/` by running `go generate` which in turn runs `go run generate.go`. It generates an AST and prints that to `raw_templates.go`.

Maybe we should `echo /raw_templates.go -diff >> .gitattributes`

Reply:
Funny that both of you said the same thing :). In sync!

Reply:
https://github.com/dxw/wpc/pull/50